whichone/static/js/data.js

```diff
@@ -96,7 +96,9 @@ async function request(key, callback) {
       user[key] = JSON.parse(data);
       user["expire"] = new Date().getTime();
       localStorage.setItem(spotify_id, JSON.stringify(user));
-      callback();
+      if (callback != null) {
+        callback();
```

Comment:
Might want to consider best way to handle async code. You are using a mixture of promises and callbacks. I think generally promises (& async/await) is preferable.